packages/amplify-graphql-index-transformer/src/graphql-index-transformer.ts

```diff
@@ -107,6 +108,19 @@ function validate(config: IndexDirectiveConfiguration, ctx: TransformerContextPr
     fieldMap.set(objectField.name.value, objectField);
 
     for (const peerDirective of objectField.directives!) {
+      if (
+        !defaultGSI &&
+        peerDirective.name.value === directiveName &&
+        objectField == config.primaryKeyField &&
+        !peerDirective.arguments!.some((arg: any) => arg.name.value === 'sortKeyFields')
```

Comment:
This validation has some overlap with the validation on lines 131-134. Is it possible to do some consolidation.

Reply:
added `hasSortFields` to move the check into its own variable.